.github/workflows/ci_frontend.yml

```diff
@@ -50,3 +50,44 @@ jobs:
           path: reports/jest-*.xml
           reporter: jest-junit
           working-directory: 'RetrospectiveExtension.Frontend'
+        
+      - if: steps.changes.outputs.src == 'true'
+        name: Get Coverage for badge
+        working-directory: 'RetrospectiveExtension.Frontend'
+        run: |
+          SUMMARY="$(npm test -- --coverageReporters='text-summary' | tail -2 | head -1)"
+          TOKENS=($SUMMARY)
+          # process.env.COVERAGE = '31.93%';
+          echo "COVERAGE=$(echo ${TOKENS[2]})" >> $GITHUB_ENV
+          # var REF = 'refs/pull/27/merge.json';
+          REF=${{ github.ref }}
+          # console.log('github.ref: ' + REF);
+          echo "github.ref: $REF"
+          # var PATHS = REF.split('/');
+          IFS='/' read -ra PATHS <<< "$REF"
+          # var BRANCH_NAME = PATHS[1] + PATHS[2];
+          BRANCH_NAME="${PATHS[1]}_${PATHS[2]}"
+          # console.log(BRANCH_NAME); // 'pull_27'
+          echo $BRANCH_NAME
+          # process.env.BRANCH = 'pull_27';
+          echo "BRANCH=$(echo ${BRANCH_NAME})" >> $GITHUB_ENV
+
+      - if: steps.changes.outputs.src == 'true'
+        name: Create the Badge
+        uses: schneegans/dynamic-badges-action@v1.0.0
+        with:
+          auth: ${{ secrets.GIST_SECRET }}
+          gistID: b414faf6b91eaaa6ec7d82882be36935
+          filename: retrotool__${{ env.BRANCH }}.json
+          label: Test Coverage
+          message: ${{ env.COVERAGE }}
+          color: green
+          namedLogo: jest
+        
+      - if: steps.changes.outputs.src == 'true'
+        name: Test Coverage PR comment
+        uses: thollander/actions-comment-pull-request@v1
+        with:
+          message: 'Test Coverage! ![Coverage Badge](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/mhassaninmsft/b414faf6b91eaaa6ec7d82882be36935/raw/retrotool__${{ env.BRANCH }}.json)'
```

Comment:
🤔 based on the comment that was generated, I don't think this is working properly. I thought that the comment is related to the PR #, not the branch name?

Reply:
Thanks for the catch, the message needed to be in double quoted to evaluate the `${{ expression }}`. The "BRANCH" tag is misleading here since it is actually the PR number or Branch name if it is a merge request

Reply:
yes we to add the `GIST_SECRET` to github secrets so that the publishing occurs. I will work with Narmatha today to get that in before merging the PR